packages/api/src/codegen/languages/typescript/index.ts

```diff
@@ -458,25 +397,115 @@ sdk.server('https://eu.api.example.com/v14');`),
     });
 
     sourceFile.addExportAssignment({
-      // Because CJS targets have `createSDK` exported with `module.exports`, but the TS type side
-      // of things to work right we need to set this as `export =`. Thankfully `ts-morph` will
-      // handle this accordingly and still create our JS file with `module.exports` and not
-      // `export =` -- only TS types will have this export style.
-      isExportEquals: this.compilerTarget === 'cjs' && this.outputJS,
+      // Because we're exporting `createSDK` as an IIFE constant we need to have it exported as
+      // `export default createSDK`. `addExportAssignment` by default wants it exported as
+      // `export = createSDK`, which will throw TS errors because we may also be exporting types in
+      // the `./types.ts` file.
+      isExportEquals: false,
       expression: 'createSDK',
     });
 
     return sourceFile;
   }
 
+  /**
+   * Create the `tsconfig.json` file that will allow this SDK to be compiled for use.
+   *
+   */
+  createTSConfig() {
+    const sourceFile = this.project.createSourceFile('tsconfig.json', '');
+
+    const config: TsConfigJson = {
+      compilerOptions: {
+        checkJs: true,
+        esModuleInterop: true,
+        module: 'NodeNext',
+        resolveJsonModule: true,
+      },
+      include: ['./src/**/*'],
+      exclude: ['dist'],
```

Comment:
is this necessary?

```suggestion
```

Reply:
Forget why I added this but it doesn't seem to be needed anymore.